crates/sui-keys/src/key_derive.rs

```diff
@@ -110,11 +153,23 @@ pub fn validate_path(
                     "m/{DERVIATION_PATH_PURPOSE_SECP256K1}'/{DERIVATION_PATH_COIN_TYPE}'/0'/0/0"
                 )
                 .parse()
-                .unwrap()),
+                .map_err(|_| SuiError::SignatureKeyGenError("Cannot parse path".to_string()))?),
             }
         }
         SignatureScheme::BLS12381 => Err(SuiError::UnsupportedFeatureError {
-            error: "BLS is not supported for user key derivation".to_string(),
+            error: "TODO".to_string(),
```

Comment:
Is the TODO the correct error term here?

Reply:
revised